sdk/keyvault/azkeys/client.go

```diff
@@ -615,60 +667,67 @@ func (s *DeleteKeyPoller) pollUntilDone(ctx context.Context, t time.Duration) (D
 		time.Sleep(t)
 	}
 
+	vaultURL, name, version := shared.ParseID(s.deleteResponse.Key.Kid)
 	return DeleteKeyResponse{
 		DeletedKey: DeletedKey{
 			RecoveryID:         s.deleteResponse.RecoveryID,
 			DeletedOn:          s.deleteResponse.DeletedDate,
 			ScheduledPurgeDate: s.deleteResponse.ScheduledPurgeDate,
-			Tags:               convertGeneratedMap(s.deleteResponse.Tags),
-			Managed:            s.deleteResponse.Managed,
 			ReleasePolicy:      keyReleasePolicyFromGenerated(s.deleteResponse.ReleasePolicy),
-			Properties:         keyPropertiesFromGenerated(s.deleteResponse.Attributes),
+			Properties:         keyPropertiesFromGenerated(s.deleteResponse.Attributes, s.deleteResponse.Key.Kid, name, version, s.deleteResponse.Managed, vaultURL, s.deleteResponse.Tags),
 			Key:                jsonWebKeyFromGenerated(s.deleteResponse.Key),
 		},
 	}, nil
 }
 
-// DeleteKeyPollerResponse contains the response from the Client.BeginDeleteKey method
-type DeleteKeyPollerResponse struct {
-	// PollUntilDone will poll the service endpoint until a terminal state is reached or an error occurs
-	PollUntilDone func(context.Context, time.Duration) (DeleteKeyResponse, error)
-
-	// Poller contains an initialized WidgetPoller
-	Poller *DeleteKeyPoller
+// ResumeToken returns a token for resuming polling at a later time
+func (s *DeleteKeyPoller) ResumeToken() (string, error) {
+	return string(s.resumeToken), nil
 }
 
 // BeginDeleteKey deletes a key from the keyvault. Delete cannot be applied to an individual version of a key. This operation
 // requires the key/delete permission. This response contains a Poller struct that can be used to Poll for a response, or the
 // PollUntilDone function can be used to poll until completion. Pass nil to use the default options.
-func (c *Client) BeginDeleteKey(ctx context.Context, keyName string, options *BeginDeleteKeyOptions) (DeleteKeyPollerResponse, error) {
+func (c *Client) BeginDeleteKey(ctx context.Context, name string, options *BeginDeleteKeyOptions) (*DeleteKeyPoller, error) {
 	if options == nil {
 		options = &BeginDeleteKeyOptions{}
 	}
-	resp, err := c.kvClient.DeleteKey(ctx, c.vaultUrl, keyName, options.toGenerated())
-	if err != nil {
-		return DeleteKeyPollerResponse{}, err
+	var resumeToken string
```

Comment:
Same feedback as on secrets, seems like the poller ought to carry the token in both cases